src/utils/getOriginalLocation.ts

```diff
@@ -4,10 +4,9 @@ export function getOriginalLocation(
 	sourcemapChain: DecodedSourceMapOrMissing[],
 	location: { column: number; line: number; name?: string; source?: string }
 ): { column: number; line: number } {
-	// This cast is guaranteed. If it were a missing Map, it wouldn't have a mappings.
 	const filteredSourcemapChain = sourcemapChain.filter(
-		sourcemap => sourcemap.mappings
-	) as ExistingDecodedSourceMap[];
+		(sourcemap): sourcemap is ExistingDecodedSourceMap => sourcemap.mappings !== undefined
```

Comment:
Are you sure you want to explicitly compare against undefined and there will never be a `null`? The "good" type should be an object, so it would be ok to also filter out e.g. empty strings etc. I think I liked the previous check better.

Reply:
@lukastaegert   uh, sorry, I think I came too late. :/

compare the original source code: `sourcemap => sourcemap.mappings` ---> no comparison at all

I addd it only because according to the types, `mappings` can be an `array` and `undefined`.

```
export type DecodedSourceMapOrMissing =
	| {
			mappings?: never;
			missing: true;
			plugin: string;
	  }
	| ExistingDecodedSourceMap;
```

> I think I liked the previous check better.

there were none. ;)

Reply:
The previous check was a check for truthiness, which you replaced with an explicit but much more strict one.

Reply:
the previous check seemed to be insufficient for typescript, otherwise I would have left it as is. 😉 

https://github.com/rollup/rollup/runs/4639006105?check_suite_focus=true

I assume that `filter` expects a `real boolean`, not just a `truthy/falsy` value. (at least in the world of TS).

do you think the typing is incorrect then? can `mappings` be null as well? presumably a check for `!= null` would be better? to check for `falsy` or `thruthy` values seems to be over checking for things. if mappings can only be an `array` or `undefined` (and possibly `null`) I don't think it's good pattern to check for anything else truthy (0, '', and what not) _just in case_, unless that case is valid. in that case tho, the typings might be incorrect.

I think it makes the code more readable and more robust. it's really hard to follow any code paths otherwise, specially if the code and the types don't match.

Reply:
I vaguely remember that mappings could be an empty string, but I am not sure in this instance. Moreover, anything could be in a user generated source map. I am just saying, the previous check was concise and worked well, also for TypeScript. TypeScript understands very well that filter does an implicit cast to Boolean and does not flag anything here. The new check might work as well, but it needlessly takes the risk of breaking stuff for other people, and I do not see a need to do so here.